### PR TITLE
User email stored to local storage after enabling sync and removed after disabling sync

### PR DESCRIPTION
**Pull request by himanish-star**
fixes; #629 
GIF of the changes:
ezgif com-video-to-gif

### CODE REVIEW

File: src/sidebar/panel.js [hunk 1]
```diff
@@ -111,12 +111,16 @@ ClassicEditor.create(document.querySelector('#editor'), {
             clearTimeout(loginTimeout);
             // set title attr of footer to the currently logged in account
             footerButtons.title = eventData.profile && eventData.profile.email;
+            localStorage.setItem('userEmail',footerButtons.title);
```
Comment (Natim) [inline src/sidebar/panel.js]: nit: add a space after the comma

Comment (himanish-star) [reply src/sidebar/panel.js]: This is a very stupid question but what is `nit`. And I'll surely make the changes

Comment (Natim) [reply src/sidebar/panel.js]: nit stand for nit-pick, it is a non really import comment.
Instead that in this case the linter might fail.

File: src/sidebar/panel.js [hunk 2]
```diff
@@ -177,6 +181,9 @@ ClassicEditor.create(document.querySelector('#editor'), {
             syncingInProcess = false;
             break;
           case 'disconnected':
+            if (localStorage.getItem('userEmail')) {
+              localStorage.removeItem('userEmail');
```
[the remark this answers is not in the document]
Comment (himanish-star) [reply src/sidebar/panel.js]: I'll make the changes in a while.  Currently I don't have my laptop 😅

File: src/sidebar/panel.js [hunk 3]
```diff
@@ -177,6 +183,7 @@ ClassicEditor.create(document.querySelector('#editor'), {
             syncingInProcess = false;
             break;
           case 'disconnected':
+            localStorage.removeItem('userEmail');
```
[the remark this answers is not in the document]
Comment (himanish-star) [reply src/sidebar/panel.js]: I have made all the changes

[the remark this answers is not in the document]
Comment (Natim) [reply src/sidebar/panel.js]: It seems to be the case here: https://github.com/mozilla/notes/pull/637/files#diff-1abf0f3e5a69394934ea8f7b5c521805R188

[the remark this answers is not in the document]
Comment (himanish-star) [reply src/sidebar/panel.js]: Yep, I had seen this but I thought as this eventually leads to the `kinto-loaded case` I thought of making the changes here

File: src/sidebar/panel.js [hunk 4]
```diff
@@ -39,6 +39,7 @@ let ignoreNextLoadEvent = false;
 let ignoreTextSynced = false;
 let lastModified;
 let lastGood = null;
+let userEmail;
```
Comment (Natim) [inline src/sidebar/panel.js]: I think we should not use a global variable here.

File: src/sidebar/panel.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Natim) [reply src/sidebar/panel.js]: Ok I understood you question, I guess the mistake is to have userEmail as a global variable.

File: src/sidebar/panel.js (hunk 4 above)
Comment (himanish-star) [reply src/sidebar/panel.js]: Done

Comment (himanish-star) [reply src/sidebar/panel.js]: @Natim , I'm so sorry in a hurry I made a forced update and missed out the review commit added by you. I'll add that back

**Comment by himanish-star** [comment]
@Natim, I guess this won't work as `let userEmail` is added somewhere else apart from where it is used. That's the reason why I was making the change. I have the commit ready shall I file a new PR now

**Comment by Natim** [comment]
No I fixed it here: 4b22218

Thanks

**Comment by vladikoff** [comment]
Thank you @himanish-star !!

**Pull request merged**